Source/Core/DolphinQt/Host.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QMutex>
+#include <QObject>
+#include <QSize>
+
+// Singleton that talks to the Core via the interface defined in Core/Host.h.
+// Because Host_* calls might come from different threads than the MainWindow,
+// the Host class communicates with it via signals/slots only.
+
+// Many of the Host_* functions are ignored, and some shouldn't exist.
```

Comment:
Sorry, that's poor communication on my part. Part of this is because I only wanted to implement what was necessary to get it to work, and part is because some of these really aren't necessary. For instance, the call to `Host_UpdateMainFrame` [here](https://github.com/dolphin-emu/dolphin/blob/dc096d2fe5408254430f2c5b31605d907ffb6901/Source/Core/Core/Core.cpp#L229) is necessary if you disable your buttons manually when your game boots, but I would much rather just connect the EmulationStarted signal with the enable/disable method of the buttons individually and let them take care of themselves from then on. More generally, the core should have no say in how I manage my GUI thank you very much.